sched/scheduler/task_scheduler.go

```diff
@@ -33,6 +36,87 @@ func getTaskAssignments(cs *clusterState, tasks []*taskState) ([]taskAssignment,
 		snapshotIds = append(snapshotIds, snapId)
 	}
 
+	// Sort jobs by priority.
```

Comment:
do we want to encapsulated this to make it easier to try out different approaches?